feature-utils/poly-import/src/importer.js

```diff
@@ -48,52 +47,94 @@ class ImporterExecutionResult {
     }
 }
 
-export async function runImporter(importerClass, zipFile, facebookAccount) {
+export async function runImporter({ importerClass, zipFile, pod, account }) {
     const importer = new importerClass();
 
     const telemetry = new Telemetry();
 
     try {
-        const status = await importer.import({
+        const response = await importer.import({
             zipFile,
-            facebookAccount,
+            pod,
+            //this is to support old importers without needing to change all at once
+            //TODO: change all importers so this can go
+            account,
+            facebookAccount: account,
         });
-        return new ImporterExecutionResult(
-            importer,
-            status,
-            telemetry.elapsedTime()
-        );
+
+        //Currently we have to do this check as not all importers return a result even when
+        //executing successfully. We can go back to destructuring after all importers have been changed
+        return {
+            report: new ImporterExecutionReport({
+                importer,
+                status: response?.status,
+                executionTime: telemetry.elapsedTime(),
+            }),
+            result: response?.result,
+            importedFileNames: response?.importedFileNames,
+        };
     } catch (error) {
-        return new ImporterExecutionResult(
-            importer,
-            new Status({ name: statusTypes.error, message: error }),
-            telemetry.elapsedTime()
-        );
+        return {
+            report: new ImporterExecutionReport({
+                importer,
+                status: new Status({ name: statusTypes.error, message: error }),
+                executionTime: telemetry.elapsedTime(),
+            }),
+        };
     }
 }
 
-export async function runImporters(importerClasses, zipFile, dataAccount) {
+export async function runImporters({ importerClasses, zipFile, account, pod }) {
     return await Promise.all(
         importerClasses.map(async (importerClass) => {
-            return runImporter(importerClass, zipFile, dataAccount);
+            return runImporter({ importerClass, zipFile, account, pod });
         })
     );
 }
 
-export async function importZip({ dataImporters, zipFile, DataAccount }) {
-    const dataAccount = new DataAccount();
-    const importingResults = await runImporters(
-        dataImporters,
-        zipFile,
-        dataAccount
-    );
+//We need this to support the tests for the previous importer model
+export async function runOutdatedImporter(
+    importerClass,
+    zipFile,
+    account,
+    pod
+) {
+    const importer = new importerClass();
 
-    dataAccount.importingResults = importingResults;
+    const telemetry = new Telemetry();
```

Comment:
Problem is, I'm starting to have a dejà vu sensation. Why don't we
1. Deal with #1190 before going on to this one.
2. Rebase this to #1190 branch and work on these if they are needed for that big one.

Reply:
Thats exactly what I am trying to do here. Therefore the **Waits for** section. By asking your review on the others I only wanted to signalise that they are all ready as soon as the first one is merged

Reply:
I didn't want to mark this as draft because it's not a draft, it's done. Maybe should not ask for your review at this point, but I wanted to signalise to get too it because they are going to pile up otherwise

Reply:
Maybe rebase to #1190 then? At least it will be much clearer what's different. You can always rebase to main when #1190 is merged.

Reply:
It's based on #1190 yet I'm merging to main or should I not?

Reply:
You're *basing* to main right now; if you rebase to #1190 we'll be able to see only the diffs from there